components/camel-kamelet/src/main/java/org/apache/camel/component/kamelet/KameletComponent.java

```diff
@@ -348,19 +362,36 @@ public LifecycleHandler() {
             this.initialized = new AtomicBoolean();
         }
 
-        public static void createRouteForEndpoint(KameletEndpoint endpoint) throws Exception {
-            LOGGER.debug("Creating route from template={} and id={}", endpoint.getTemplateId(), endpoint.getRouteId());
+        public void createRouteForEndpoint(KameletEndpoint endpoint) throws Exception {
+            final ExtendedCamelContext ecc = getCamelContext().adapt(ExtendedCamelContext.class);
+            final ModelCamelContext context = getCamelContext().adapt(ModelCamelContext.class);
+            final String templateId = endpoint.getTemplateId();
+            final String routeId = endpoint.getRouteId();
+
+            if (context.getRouteTemplateDefinition(templateId) == null) {
+                LOGGER.debug("Loading route template={} from {}", templateId, getLocation());
+
+                String path = getLocation();
+                if (path != null) {
+                    if (!path.endsWith("/")) {
+                        path += "/";
+                    }
+
+                    ecc.getRoutesLoader().loadRoutes(
+                            ecc.getResourceLoader().resolveResource(path + templateId + ".kamelet.yaml"));
```

Comment:
I think we may need a blurb in the documentation (somewhere) about the convention to use templateId.kamelet.yaml as the default file pattern for loading kamelets.